hazelcast/src/main/java/com/hazelcast/map/impl/query/PartitionScanRunner.java

```diff
@@ -87,14 +88,17 @@ public void run(String mapName, Predicate predicate, int partitionId, Result res
         while (iterator.hasNext()) {
             Record record = iterator.next();
             Data key = (Data) toData(record.getKey());
+            Metadata metadata = record.getMetadata();
             Object value = toData(
                     useCachedValues ? Records.getValueOrCachedValue(record, serializationService) : record.getValue());
             if (value == null) {
                 continue;
             }
 
             queryEntry.init(serializationService, key, value, extractors);
-            if (predicate.apply(queryEntry) && compareAnchor(pagingPredicate, queryEntry, nearestAnchorEntry)) {
+            queryEntry.setMetadata(metadata);
```

Comment:
to make it less error prone, can we pass this metadata with `init` method?

Reply:
I can try to refactor this in a followup PR. Changing the init method triggers a lot of changes. I do not want to pollute this PR with those unrelated changes.